src/data/transforms/minmax.rs

```diff
@@ -0,0 +1,247 @@
+//! The Min-Max transformer
+//!
+//! This module contains the `MinMaxScaler` transformer.
+//!
+//! The `MinMaxScaler` transformer is used to transform input data
+//! so that the minimum and maximum of each column are as specified.
+//! This is commonly used to transform the data to have a minimum of
+//! `0` and a maximum of `1`.
+//!
+//! # Examples
+//!
+//! ```
+//! use rusty_machine::data::transforms::{Transformer, MinMaxScaler};
+//! use rusty_machine::linalg::Matrix;
+//!
+//! // Constructs a new `MinMaxScaler` to map minimum to 0 and maximum
+//! // to 1.
+//! let mut transformer = MinMaxScaler::default();
+//!
+//! let inputs = Matrix::new(2, 2, vec![-1.0, 2.0, 1.5, 3.0]);
+//!
+//! // Transform the inputs to get output data with required minimum
+//! // and maximum.
+//! let transformed = transformer.transform(inputs).unwrap();
+//! ```
+
+use learning::error::{Error, ErrorKind};
+use linalg::Matrix;
+use super::Transformer;
+
+use libnum::Float;
+
+/// The `MinMaxScaler`
+///
+/// The `MinMaxScaler` provides an implementation of `Transformer`
+/// which allows us to transform the input data to have a new minimum
+/// and maximum per column.
+///
+/// See the module description for more information.
+#[derive(Debug)]
+pub struct MinMaxScaler<T: Float> {
+    /// Mins per column of input data
+    scale_factors: Option<Vec<T>>,
+    /// Maxs per column of input data
+    const_factors: Option<Vec<T>>,
```

Comment:
Thanks for reading this over.

This is an artifact of me changing the name of the field. I'll update now, thanks! (These comments wont be visible in docs as the fields aren't public - but still well worth getting right).